resource-managers/mesos/src/main/scala/org/apache/spark/scheduler/cluster/mesos/MesosClusterScheduler.scala

```diff
@@ -542,13 +542,23 @@ private[spark] class MesosClusterScheduler(
       "spark.master" // this contains the address of the dispatcher, not master
     )
     val defaultConf = conf.getAllWithPrefix("spark.mesos.dispatcher.driverDefault.").toMap
+    println("L545 desc.conf.getAll.foreach -> printing all confs before shell-escape")
+    println("================================================================")
+    desc.conf.getAll.foreach(println)
+
     val driverConf = desc.conf.getAll
       .filter { case (key, _) => !replicatedOptionsBlacklist.contains(key) }
       .toMap
     (defaultConf ++ driverConf).foreach { case (key, value) =>
-      options ++= Seq("--conf", s""""$key=${shellEscape(value)}"""".stripMargin) }
-
-    options
+      options ++= Seq("--conf", s"${key}=${value}") }
+    println("L554: options -> printing all options after key=value set")
+    println("================================================================")
+
+    val opts = options.map(shellEscape)
```

Comment:
Can we add a test for this? Also please remove the printlns

Reply:
Was just a test, planning to delete the last commit

Reply:
Doesn't look like Apache/Spark caught the issue for `--py-files`. Change+test added in [9f110fd](https://github.com/mesosphere/spark/pull/33/commits/9f110fd465b43cf255dc2b6306391ecf0ea19eb5)